src/target/cortexar.c

```diff
@@ -1171,7 +1223,13 @@ static inline bool cortexar_mem_write_fast(target_s *const target, const uint32_
 
 	/* Write each of the uint32_t's checking for failure */
 	for (size_t offset = 0; offset < count; ++offset) {
-		if (!cortexar_run_write_insn(target, ARM_STC_DTRRX_R0_POSTINC4_INSN, src[offset]))
+		uint32_t value;
+		if (target->target_options & TOPT_FLAVOUR_BE) {
+			value = read_be4((const void *const)src, offset * 4);
+		} else {
+			value = read_le4((const void *const)src, offset * 4);
+		}
```

Comment:
The braces here are superfluous and can be dropped. We would be surprised if `src` needs casing at all, it should automatically cast to `void *` and get `const` added as needed (removal of const requires casts, not addition).

Please suffix `4` with `U` to make the multiplication unsigned.

Reply:
The cast is required because the `read_*` functions take a `const uint8_t *const` and not `void *`:

```
../src/target/cortexar.c:1229:21: warning: incompatible pointer types passing 'const uint32_t *const' (aka 'const unsigned int *const') to parameter of type 'const uint8_t *' (aka 'const unsigned char *') [-Wincompatible-pointer-types]
 1229 |                         value = read_be4(src, offset * 4);
      |                                          ^~~
../src/include/buffer_utils.h:97:54: note: passing argument to parameter 'buffer' here
   97 | static inline uint32_t read_be4(const uint8_t *const buffer, const size_t offset)
      |                                                      ^
```

Reply:
In that case, you can simplify the cast to just `(const void *)` - the compiler will add the final `const` for us.